tests/client/ui/components/TestLoadingIndicator.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Simulate, renderIntoDocument } from 'react-addons-test-utils';
+
+import LoadingIndicator from 'ui/components/LoadingIndicator';
+
+
+describe('<LoadingIndicator />', () => {
+  function render(props) {
+    return renderIntoDocument(<LoadingIndicator {...props} />);
+  }
+
+  it('resets the animation after it ends', (done) => {
+    const root = render({ timeToReset: 5 });
+    const iconNode = ReactDOM.findDOMNode(root.loadingIcon);
+
+    assert.isTrue(root.state.animating);
+
+    Simulate.animationEnd(iconNode);
+
+    assert.isFalse(root.state.animating);
+
+    setTimeout(() => {
+      assert.isTrue(root.state.animating);
+      done();
+    }, 10);
```

Comment:
As @mstriemer mentioned seems it can be done in CSS; I'm just fixing up another PR then gonna do that here. Should tidy it up a million.